packages/wonder-blocks-popover/components/popover-dialog.js

```diff
@@ -0,0 +1,124 @@
+// @flow
+import * as React from "react";
+import {StyleSheet} from "aphrodite";
+
+import {View} from "@khanacademy/wonder-blocks-core";
+import {TooltipTail} from "@khanacademy/wonder-blocks-tooltip";
+import Color from "@khanacademy/wonder-blocks-color";
+
+import type {
+    Placement,
+    PopperElementProps,
+} from "@khanacademy/wonder-blocks-tooltip";
+
+import PopoverContent from "./popover-content.js";
+import PopoverContentCore from "./popover-content-core.js";
+
+type Props = {|
+    /**
+     * Required to correctly position the elements inside the dialog
+     * @ignore
+     */
+    ...PopperElementProps,
+
+    /**
+     * The content to render inside the dialog.
+     */
+    children:
+        | React.Element<typeof PopoverContent>
+        | React.Element<typeof PopoverContentCore>,
+
+    /**
+     * Called when popper changes its placement
+     */
+    onUpdate: (placement: Placement) => mixed,
+|};
+
+/**
+ * This is an internal component that we use to render the stuff that appears
+ * when a popover shows. It's composed by two elements: The popover content,
+ * that can be of type [PopoverContent](#PopoverContent) or
+ * [PopoverContentCore](#PopoverContentCore), and the
+ * [TooltipTail](#TooltipTail).
+ *
+ * Note that without explicit positioning, the tail will not be centered.
+ */
+export default class PopoverDialog extends React.Component<Props> {
+    componentDidUpdate(prevProps: Props) {
+        if (prevProps.placement !== this.props.placement) {
+            this.props.onUpdate(this.props.placement);
+        }
```

Comment:
Can you add a comment on why we're doing this? I think I get it, but it will be useful for future code viewers to understand why we have this.